staging/src/k8s.io/client-go/features/env_var_feature_gate.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"sync/atomic"
+
+	"k8s.io/apimachinery/pkg/util/naming"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+)
+
+// internalPackages are packages that ignored when creating a name for featureGates. These packages are in the common
+// call chains, so they'd be unhelpful as names.
+var internalPackages = []string{"k8s.io/client-go/features/env_var_feature_gate.go"}
+
+var _ Reader = &envVarFeatureGate{}
+
+// newEnvVarFeatureGate returns a new Reader that recognizes the provided features.
+//
+// On the first call to Enabled, the effective state of all known features is loaded from
+// environment variables. The environment variable read for a given feature is formed by
+// concatenating the prefix "KUBE_FEATURE_" with the feature's name.
+//
+// For example, if you have a feature named "MyFeature,"
+// setting an environmental variable "KUBE_FEATURE_MyFeature"
+// will allow you to configure the state of that feature.
+//
+// Please note that environmental variables can only be set to a boolean type.
+// Incorrect values will be ignored and logged.
+func newEnvVarFeatureGate(features map[Feature]FeatureSpec) *envVarFeatureGate {
+	enabled := map[Feature]bool{}
+	enabledValue := &atomic.Value{}
+	enabledValue.Store(enabled)
+
+	known := map[Feature]FeatureSpec{}
+	for name, spec := range features {
+		known[name] = spec
+	}
+
+	return &envVarFeatureGate{
+		envVarFeatureGateName: naming.GetNameFromCallsite(internalPackages...),
+		known:                 known,
+		enabled:               enabledValue,
+	}
+}
+
+// envVarFeatureGate implements Reader and allows for feature registration.
+type envVarFeatureGate struct {
+	// envVarFeatureGateName holds the name of the file
+	// that created this instance
+	envVarFeatureGateName string
```

Comment:
this stutters. Why the prefix?

Reply:
why not say what it is: sourceFileName ?

Reply:
right, will remove the prefix.